### PR TITLE
Go: Update to the latest revision of Go 1.11 on Linux

### PR DESCRIPTION
**Pull request by Caiyeon**
For linux images: changes go version from 1.11 to 1.11.2

Minor revisions 1.11.2 includes changes: 
https://golang.org/doc/devel/release.html#go1.11.minor
https://github.com/golang/go/issues?q=milestone%3AGo1.11.2

Which also includes changes for 1.11.1:
https://github.com/golang/go/issues?q=milestone%3AGo1.11.1

### CODE REVIEW

File: images/linux/scripts/installers/go.sh [hunk 1]
```diff
@@ -32,4 +32,4 @@ function InstallGo () {
 # Install Go versions
 InstallGo 1.9 1.9.7 false
 InstallGo 1.10 1.10.4 false
```
Comment (bishal-pdMSFT) [inline images/linux/scripts/installers/go.sh]: The title says that Go1.10 is also updated, but it is still 1.10.4. Do you want to update it to 1.10.5?

**Comment by bishal-pdMSFT** [comment]
This PR actually only updates linux image as opposed to the title which says windows as well. https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/win/scripts/Installers/Install-Go.ps1 should also be updated for windows.

File: images/linux/scripts/installers/go.sh (hunk 1 above)
Comment (Caiyeon) [reply images/linux/scripts/installers/go.sh]: The title was edited by @DavidStaheli 

The original title was: `Update minor revision of go for 1.11`

I am indifferent to updating go 1.10

**Comment by Caiyeon** [comment]
> This PR actually only updates linux image as opposed to the title which says windows as well. https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/win/scripts/Installers/Install-Go.ps1 should also be updated for windows.

The title was updated by @DavidStaheli 

I am indifferent to updating go for windows images, but perhaps they should be in a separate PR?

I am only personally interested in linux images

File: images/linux/scripts/installers/go.sh (hunk 1 above)
Comment (bishal-pdMSFT) [reply images/linux/scripts/installers/go.sh]: Updated title accordingly

**Pull request merged**